src/main.rs

```diff
@@ -1,5 +1,9 @@
 use std::io;
 
+fn count_overlap(secret: &str, guess: &str) -> usize {
+    secret.chars().filter(|letter| guess.contains(*letter)).count()
```

Comment:
I like pulling this functionality out into a separate function. Nice use of rust built-in functions.

Just slightly curious if this new function is more performant though?